kenlm_training/train_all.sh

```diff
@@ -0,0 +1,134 @@
+REMOVE_ACCENTS=False
+LOWER_CASE=False
+NORMALIZE_NUMBERS=True
+NORMALIZE_PUNCT=1
+
+# OSCAR
+NDOC_FOR_LM_OSCAR=1_000_000
+
+
+train_language_and_dataset () {
+    local lang=$1
+    local dataset=$2
+    if [ $dataset = "wikipedia" ]; then
+        # 1 Download Wikipedia cirrus
+        if [ -f "data/${dataset}/cirrus/gz/${lang}.json.gz" ]; then
+            echo "${lang} Wikipedia cirrus was already downloaded."
+        else
+            echo "Downloading ${lang}"
+            mkdir -p data/${dataset}/cirrus/gz/
+            python cc_net/get_wiki_cirrus.py dl --lang ${lang} --output_dir data/${dataset}/cirrus/gz --date 20211115
+            echo "Downloaded Wikipedia cirrus for ${lang}"
+        fi
+
+        # 2 Extract opening text of each article
+        if [ -f "data/${dataset}/cirrus/gz/${lang}.opening.txt" ]; then
+            echo "Wikipedia openings were already extracted for ${lang}"
+        else
+            echo "Extracting ${lang}"
+            python cc_net/get_wiki_cirrus.py opening \
+                --n_docs ${NDOC_FOR_LM} \
+                --file "data/${dataset}/cirrus/gz/${lang}.json.gz" \
+                --output "data/${dataset}/cirrus/gz/${lang}.opening.txt" \
+                --accent ${REMOVE_ACCENTS} \
+                --case ${LOWER_CASE} \
+                --numbers ${NORMALIZE_NUMBERS} \
+                --punct ${NORMALIZE_PUNCT}
+        fi
+    else
+        # 1 & 2 Download and preprocess dataset from HF hub
+        if [ -f "data/${dataset}/cirrus/gz/${lang}.opening.txt" ]; then
+            echo "Wikipedia openings were already extracted for ${lang}"
+        else
+            echo "Downloading OSCAR ${lang}"
+            mkdir -p "data/${dataset}/cirrus/gz/"
+            python cc_net/get_hf_dataset.py dl \
+                --dataset ${dataset} \
+                --output_file "data/${dataset}/cirrus/gz/${lang}.opening.txt" \
+                --name "unshuffled_deduplicated_${lang}" \
+                --split "train" \
+                --max_docs $NDOC_FOR_LM_OSCAR
+        fi
+    fi
+
+    # 3 Train sentence piece tokenizer
+    if [ -f "data/${dataset}/lm_sp/${lang}.sp.model" ]; then
+        echo "Sentence piece tokenizer was already trained for ${lang}"
+    else
+        echo "Training sentence piece tokenizer for ${lang}"
+        mkdir -p data/${dataset}/lm_sp
+        ./bin/spm_train --input="data/${dataset}/cirrus/gz/${lang}.opening.txt" \
+            --vocab_size=${VOCAB_SIZE} --hard_vocab_limit \
+            --character_coverage=0.9995 \
+            --model_type=unigram \
+            --model_prefix="data/${dataset}/lm_sp/${lang}.sp" \
+        || echo "WARNING: Corpus is too small, will train smaller model" && \
+        ./bin/spm_train --input="data/${dataset}/cirrus/gz/${lang}.opening.txt" \
+            --vocab_size=${SMALL_VOCAB_SIZE} \
+            --character_coverage=0.9995 \
+            --model_type=unigram \
+            --model_prefix="data/${dataset}/lm_sp/${lang}.sp"
```

Comment:
Thanks @versae, great catch! I think we should just keep the second line then, rather than fixing the bug, for reproducibility - since these models were used to generate the final SP tokenizers for the KenLM models. What do you think?

Reply:
In my short experiments, the KenLM model doesn't really change that much from one size to the other. But if the tokenizers are going to be used for training models, it'd be better to adapt the code so there are no surprises for anyone using them. My proposal, set `VOCAB_SIZE` to `40000` and remove the line training on `SMALL_VOCAB_SIZE`.